modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/LinkedProductResourceImpl.java

```diff
@@ -0,0 +1,196 @@
+import com.liferay.commerce.account.service.CommerceAccountLocalService;
+import com.liferay.commerce.account.util.CommerceAccountHelper;
+import com.liferay.commerce.product.model.CProduct;
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.permission.CommerceProductViewPermission;
+import com.liferay.commerce.product.service.CProductLocalService;
+import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.commerce.product.type.grouped.constants.GroupedCPTypeConstants;
+import com.liferay.commerce.product.type.grouped.model.CPDefinitionGroupedEntry;
+import com.liferay.commerce.product.type.grouped.service.CPDefinitionGroupedEntryService;
+import com.liferay.commerce.shop.by.diagram.constants.CSDiagramCPTypeConstants;
+import com.liferay.commerce.shop.by.diagram.model.CSDiagramEntry;
+import com.liferay.commerce.shop.by.diagram.service.CSDiagramEntryService;
+import com.liferay.headless.commerce.delivery.catalog.dto.v1_0.LinkedProduct;
+import com.liferay.headless.commerce.delivery.catalog.dto.v1_0.Product;
+import com.liferay.headless.commerce.delivery.catalog.internal.dto.v1_0.converter.LinkedProductDTOConverter;
+import com.liferay.headless.commerce.delivery.catalog.internal.dto.v1_0.converter.LinkedProductDTOConverterContext;
+import com.liferay.headless.commerce.delivery.catalog.resource.v1_0.LinkedProductResource;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.fields.NestedField;
+import com.liferay.portal.vulcan.fields.NestedFieldId;
+import com.liferay.portal.vulcan.fields.NestedFieldSupport;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import java.util.List;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(
+	properties = "OSGI-INF/liferay/rest/v1_0/linked-product.properties",
+	scope = ServiceScope.PROTOTYPE,
+	service = {LinkedProductResource.class, NestedFieldSupport.class}
+)
+public class LinkedProductResourceImpl
+	extends BaseLinkedProductResourceImpl implements NestedFieldSupport {
+
+	@NestedField(parentClass = Product.class, value = "linkedProducts")
+	@Override
+	public Page<LinkedProduct> getChannelProductLinkedProductsPage(
+			@NestedFieldId(value = "channelId") Long channelId,
+			@NestedFieldId(value = "productId") Long productId, Long accountId,
+			Pagination pagination)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannel(channelId);
+
+		long accountId1 = _getAccountId(accountId, commerceChannel);
```

Comment:
this name looks wrong @alessio-rendina can you check this?

Reply:
@alessiorendina fyi

Reply:
No need to create the variable here

Reply:
You can just reassign accountId

Reply:
`long selectedAccountId = _getSelectedAccountId(accountId, commerceChannel);`

Reply:
Done